spotify.go

```diff
@@ -0,0 +1,286 @@
+package sync
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"math"
+	"net/http"
+	"os"
+	"regexp"
+	//"sort"
+	"strings"
+
+	"github.com/zmb3/spotify"
+	"golang.org/x/oauth2"
+)
+
+var (
+	ErrNotReady = fmt.Errorf("client is not ready, make sure user has authenticated the app")
+	ErrNoEnv    = fmt.Errorf("client key and secret is not set in environment variables")
+
+	scopes = []string{spotify.ScopePlaylistModifyPublic, spotify.ScopeUserReadPrivate}
+)
+
+type Spotify struct {
+	Done chan bool
+
+	auth   spotify.Authenticator
+	client spotify.Client
+	state  string
+	ready  bool
+}
+
+const (
+	ClientKey    = "SPOTIFY_ID"
+	ClientSecret = "SPOTIFY_SECRET"
+	AuthURL      = "https://accounts.spotify.com/authorize"
+	TokenURL     = "https://accounts.spotify.com/api/token"
+)
+
+func NewSpotify(redirectURI, state string) (*Spotify, error) {
+	if os.Getenv(ClientKey) == "" && os.Getenv(ClientSecret) == "" {
```

Comment:
As per discussion, you can remove usage of Env by setting `s.SetAuthInfo(key, secret)` later.

Reply:
#4